### PR TITLE
Add cmake_common_scripts

### PR DESCRIPTION
**Pull request by Levi-Armstrong**

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@mpowelson Ready for review.

File: include/opw_kinematics/opw_kinematics.h [hunk 1]
```diff
@@ -1,12 +1,11 @@
 #ifndef OPW_KINEMATICS_H
 #define OPW_KINEMATICS_H
 
-#include <Eigen/Dense>
-#include "opw_kinematics/opw_parameters.h"
+#include <Eigen/Dense>                      // IWYU pragma: export
```
Comment (mpowelson) [inline include/opw_kinematics/opw_kinematics.h:4]: Eigen seems like a type you wouldn't want to export. If someone is using Eigen in their file, they should include it themselves and not rely on this header.

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,24 +2,32 @@ cmake_minimum_required(VERSION 3.5.0)
 project(opw_kinematics VERSION 0.1.0 LANGUAGES CXX)
 
 find_package(Eigen3 REQUIRED)
+find_package(cmake_common_scripts REQUIRED)
+
+add_code_coverage_all_targets()
+
 if(NOT EIGEN3_INCLUDE_DIRS)
   set(EIGEN3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
 endif()
```
Comment (mpowelson) [inline CMakeLists.txt:11]: What is this? You're allowed to overwrite the version of Eigen found by CMake?

File: CMakeLists.txt [hunk 3]
```diff
@@ -2,24 +2,32 @@ cmake_minimum_required(VERSION 3.5.0)
...
 
-list(FIND CMAKE_CXX_COMPILE_FEATURES cxx_std_11 CXX_FEATURE_FOUND)
+if(OPW_ENABLE_TESTING)
```
Comment (mpowelson) [inline CMakeLists.txt:13]: Maybe a comment of what you are doing here for the mere mortals that will follow you.

File: CMakeLists.txt [hunk 4]
```diff
@@ -30,17 +38,22 @@ target_include_directories(${PROJECT_NAME} SYSTEM INTERFACE
 
 # Create test executable
 add_executable(${PROJECT_NAME}_node src/test.cpp)
+target_link_libraries(${PROJECT_NAME}_node PRIVATE ${PROJECT_NAME})
 set_target_properties(${PROJECT_NAME}_node PROPERTIES OUTPUT_NAME node PREFIX "")
-target_link_libraries(${PROJECT_NAME}_node ${PROJECT_NAME})
+target_cxx_version(${PROJECT_NAME}_node PUBLIC VERSION 11)
```
Comment (mpowelson) [inline CMakeLists.txt:43]: private?

File: CMakeLists.txt [hunk 5]
```diff
@@ -2,24 +2,32 @@ cmake_minimum_required(VERSION 3.5.0)
 project(opw_kinematics VERSION 0.1.0 LANGUAGES CXX)
 
 find_package(Eigen3 REQUIRED)
+find_package(cmake_common_scripts REQUIRED)
+
+add_code_coverage_all_targets()
```
Comment (mpowelson) [inline CMakeLists.txt:7]: This probably (maybe) overwrites your enable flags in the target_add_code_coverage?

File: CMakeLists.txt (hunk 2 above)
Comment (Levi-Armstrong) [reply CMakeLists.txt:11]: I believe this is just making sure the variable EIGEN3_INCLUDE_DIRS exists and if not it sets it to EIGEN3_INCLUDE_DIR. They should be the same but I believe older versions may have only had EIGEN3_INCLUDE_DIR which was an error in Eigen.

File: CMakeLists.txt (hunk 3 above)
Comment (Levi-Armstrong) [reply CMakeLists.txt:13]: fixed.

File: CMakeLists.txt (hunk 4 above)
Comment (Levi-Armstrong) [reply CMakeLists.txt:43]: Fixed

File: CMakeLists.txt (hunk 5 above)
Comment (Levi-Armstrong) [reply CMakeLists.txt:7]: Fixed

File: include/opw_kinematics/opw_kinematics.h (hunk 1 above)
Comment (Levi-Armstrong) [reply include/opw_kinematics/opw_kinematics.h:4]: fixed.

**Comment by Levi-Armstrong** [comment]
@mpowelson I fixed the requested changes. Also I create tag v0.1.0 and increased the version to 0.2.0 since this adds a news dependency.

**Comment by mpowelson** [comment]
Ok. It looks good to me.

**Pull request merged**